sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -1324,3 +1307,98 @@ pub struct CredentialProvider {
     image_patterns: Vec<SingleLineString>,
     cache_duration: Option<KubernetesDurationValue>,
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// KubernetesMemoryReservationKey represents a string that contains a valid Kubernetes memory
+/// resource reservation key.
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct KubernetesMemoryReservationKey {
+    inner: String,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Deserialize)]
+enum MemoryReservationResources {
+    #[serde(rename = "memory")]
+    Memory,
+    #[serde(rename = "hugepages-2Mi")]
+    HugePages2Mi,
+    #[serde(rename = "hugepages-1Gi")]
+    HugePages1Gi,
+}
```

Comment:
```suggestion
/// KubernetesMemoryReservationKey represents a string that contains a valid Kubernetes memory
/// resource reservation key.
#[derive(Debug, Clone, Eq, PartialEq, Hash, Deserialize, Scalar)]
enum KubernetesMemoryReservationKey {
    #[serde(rename = "memory")]
    Memory,
    #[serde(rename = "hugepages-2Mi")]
    HugePages2Mi,
    #[serde(rename = "hugepages-1Gi")]
    HugePages1Gi,
}
```

Again for better rendering:

```rust
/// KubernetesMemoryReservationKey represents a string that contains a valid Kubernetes memory
/// resource reservation key.
#[derive(Debug, Clone, Eq, PartialEq, Hash, Deserialize, Scalar)]
enum KubernetesMemoryReservationKey {
    #[serde(rename = "memory")]
    Memory,
    #[serde(rename = "hugepages-2Mi")]
    HugePages2Mi,
    #[serde(rename = "hugepages-1Gi")]
    HugePages1Gi,
}
```